### DISCUSSION TITLE
Allow setting OPERATOR_IMAGE_NAME from a ConfigMap

### DISCUSSION THREAD

**Discussion by vanakema** [cloudnative-pg/cloudnative-pg#2080] [Ideas]
Usecase: You are working on an airgapped k8s cluster, with an intranet container repo. This means none of your image paths will match exactly what is in the default release. Your choices are now to `sed` the file, adding complexity to your build system, or use a mutating webhook, which adds complexity to your environment.

**Comment by sxd** [comment]
hi @vanakema 
Isn't enought to use the env variables you already have in the operator deployment? the operator image it's something you actually need when deploying, and ConfigMaps are mean to mutate in the future, something that cannot happen with the operator image, not in an easy way at least.

Now, if you're in an air gaped installation, it's kind of expected that you will have to change a couple of things in the deployment for sure, how using a ConfigMaps it's different from the EnvVariables here?

I'm turning this into a discussion since this doesn't look like an issue right now, or at least we can create one with better information.

Regards!

**Reply by vanakema** [reply]
We ended up using cdk8s-kustomize to change this variable. Was there a different way that we could change the `OPERATOR_IMAGE_NAME`? It would be preferable to not have to use cdk8s-kustomize
